Automattic/wp-calypso
# Storage Add Ons:  Implement dropdown checkout functionality

Related to https://github.com/Automattic/wp-calypso/pull/79041#discussion_r1279224891
Issue https://github.com/Automattic/martech/issues/2025
Tracking https://github.com/Automattic/martech/issues/2023

## Proposed Changes

* Update checkout cart for `/start` signup onboarding flow with storage add ons

## GIFs
![2023-08-31 16 10 43](https://github.com/Automattic/wp-calypso/assets/5414230/157f43f0-6f5c-458c-b822-93fb922cf61d)

## To be left as a Follow Up for the Future
* Make the storage add-on in checkout cart a dropdown ( to allow user to switch between storage add-ons ) @saygunnyc has stated that this is something we can build on iteratively

## Testing Instructions



* Smoke test flows:  in PCYsg-RDT-p2 and ensure that a free plan ( if available ) and a paid plan can be purchased
  * /setup/newsletter ( Stepper flows should be unaffected by our changes )
    * navigate to http://calypso.localhost:3000/setup/newsletter as a logged in user
    * Complete onboarding steps until you’ve reached the plans page 
  * /start/domains
    * Navigate to http://calypso.localhost:3000/start/domains
    * Select a custom domain like .org, .blog, etc.
    * Navigate to the plans page
  * /start/hosting
    * Navigate to http://calypso.localhost:3000/start/hosting
    * Navigate to the plans page
  * /start/import
    * Navigate to http://calypso.localhost:3000/start/import
    * Select any domain, free or custom
    * Navigate to the plans page
  * /start/onboarding-pm
    * Navigate to http://calypso.localhost:3000/start/onboarding-pm
    * Select any domain, free or custom
    * Navigate to the plans page
  * /start/onboarding-with-email
    * Navigate to http://calypso.localhost:3000/start/onboarding-with-email
    * Select any domain, free or custom
    * Select the provided email option
    * Navigate to the plans page
  * /start/with-themes
    * Go to [https://wordpress.com/themes while](http://calypso.localhost:3000/themes%C2%A0while) logged out and pick a Premium theme.
    * Alternatively, copy the URL of the “Pick this design” button and paste it in a logged-in browser session (it should be something like wordpress.com/start/with-theme?theme=yuga&premium=true&theme_type=premium)
    * Proceed with the onboarding until you get to the “Choose a plan” step
  * /start/personal
    * Navigate to http://calypso.localhost:3000/start/personal
    * Select any domain, free or custom
  * /start/personal-monthly
    * Navigate to http://calypso.localhost:3000/start/personal
    * Select any domain, free or custom
  * /start/personal-2y
    * Navigate to http://calypso.localhost:3000/start/personal
    * Select any domain, free or custom
  * /start/business
    * Navigate to http://calypso.localhost:3000/start/business
    * Select any domain, free or custom
  * /setup/wooexpress ( since wooexpress has no onboarding plans page, it should be unaffected by our changes )
    * Navigate to http://calypso.localhost:3000/setup/wooexpress ( If testing in a dev environment, make sure to navigate to the wordpress.com URL above to create the initial site. Then switch over to a dev URL. For some reason, using a dev URL to initially set up wooexpress doesn’t work as expected )
    * Wait until redirected to the wp-admin page for the newly created atomic site
    * From the wp-admin page click, on Upgrades > Plans
    * Alternatively, navigate to http://calypso.localhost:3000/plans/{SITE_SLUG}

## Pre-merge Checklist
